extensions/mqtt/processors/ConsumeMQTT.cpp

```diff
@@ -34,111 +34,325 @@ void ConsumeMQTT::initialize() {
+  }
 }
 
 void ConsumeMQTT::checkProperties() {
-  if (!cleanSession_ && clientID_.empty()) {
-    throw Exception(PROCESS_SCHEDULE_EXCEPTION, "Processor must have a Client ID for durable (non-clean) sessions");
+  if (mqtt_version_ == MqttVersions::V_3_1_0 || mqtt_version_ == MqttVersions::V_3_1_1 || mqtt_version_ == MqttVersions::V_3X_AUTO) {
+    if (isPropertyExplicitlySet(CleanStart)) {
+      logger_->log_warn("MQTT 3.x specification does not support Clean Start. Property is not used.");
+    }
+    if (isPropertyExplicitlySet(SessionExpiryInterval)) {
+      logger_->log_warn("MQTT 3.x specification does not support Session Expiry Intervals. Property is not used.");
+    }
+    if (isPropertyExplicitlySet(AttributeFromContentType)) {
+      logger_->log_warn("MQTT 3.x specification does not support Content Types and thus attributes cannot be created from them. Property is not used.");
+    }
+    if (isPropertyExplicitlySet(TopicAliasMaximum)) {
+      logger_->log_warn("MQTT 3.x specification does not support Topic Alias Maximum. Property is not used.");
+    }
+    if (isPropertyExplicitlySet(ReceiveMaximum)) {
+      logger_->log_warn("MQTT 3.x specification does not support Receive Maximum. Property is not used.");
+    }
+  }
+
+  if (mqtt_version_.value() == MqttVersions::V_5_0 && isPropertyExplicitlySet(CleanSession)) {
+    logger_->log_warn("MQTT 5.0 specification does not support Clean Session. Property is not used.");
+  }
+
+  if (clientID_.empty()) {
+    if (mqtt_version_.value() == MqttVersions::V_5_0) {
+      if (session_expiry_interval_ > std::chrono::seconds(0)) {
+        throw Exception(PROCESS_SCHEDULE_EXCEPTION, "Processor must have a Client ID for durable (Session Expiry Interval > 0) sessions");
+      }
+    } else if (!clean_session_) {
+      throw Exception(PROCESS_SCHEDULE_EXCEPTION, "Processor must have a Client ID for durable (non-clean) sessions");
+    }
+  }
+
+  if (qos_ == MqttQoS::LEVEL_0) {
+    if (mqtt_version_.value() == MqttVersions::V_5_0) {
+      if (session_expiry_interval_ > std::chrono::seconds(0)) {
+        logger_->log_warn("Messages are not preserved during client disconnection "
+                          "by the broker when QoS is less than 1 for durable (Session Expiry Interval > 0) sessions. Only subscriptions are preserved.");
+      }
+    } else if (!clean_session_) {
+      logger_->log_warn("Messages are not preserved during client disconnection "
+                        "by the broker when QoS is less than 1 for durable (non-clean) sessions. Only subscriptions are preserved.");
+    }
+  }
+}
+
+void ConsumeMQTT::checkBrokerLimitsImpl() {
+  auto hasWildcards = [] (std::string_view topic) {
+    return std::any_of(topic.begin(), topic.end(), [] (const char ch) {return ch == '+' || ch == '#';});
+  };
+
+  if (wildcard_subscription_available_.has_value() && !*wildcard_subscription_available_ && hasWildcards(topic_)) {
```

Comment:
I'm not sure if it would be better or not to use direct comparison to `false` here? `wildcard_subscription_available_ == false`

Reply:
It would make sense. Corrected.